src/thunderbots/software/test/world/team.cpp

```diff
@@ -0,0 +1,396 @@
+                          AngularVelocity::zero(), current_time);
+
+    std::vector<Robot> robot_list = {robot_0, robot_1};
+
+    team.updateRobots(robot_list, current_time);
+
+    EXPECT_EQ(2, team.size());
+    EXPECT_EQ(robot_0, team.getRobotById(0));
+    EXPECT_EQ(robot_1, team.getRobotById(1));
+    EXPECT_EQ(std::nullopt, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.goalie());
+    EXPECT_EQ(robot_list, team.getAllRobots());
+
+    team.clearAllRobots();
+
+    EXPECT_EQ(0, team.size());
+    EXPECT_EQ(std::nullopt, team.getRobotById(0));
+    EXPECT_EQ(std::nullopt, team.getRobotById(1));
+    EXPECT_EQ(std::nullopt, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.goalie());
+    EXPECT_EQ(std::vector<Robot>(), team.getAllRobots());
+}
+
+TEST_F(TeamTest, assign_and_clear_goalie)
+{
+    Team team = Team(milliseconds(1000));
+
+    Robot robot_0 = Robot(0, Point(0, 1), Vector(-1, -2), Angle::half(),
+                          AngularVelocity::threeQuarter(), current_time);
+
+    Robot robot_1 = Robot(1, Point(3, -1), Vector(), Angle::zero(),
+                          AngularVelocity::zero(), current_time);
+
+    std::vector<Robot> robot_list = {robot_0, robot_1};
+
+    team.updateRobots(robot_list, current_time);
+
+    EXPECT_EQ(std::nullopt, team.goalie());
+
+    team.clearGoalie();
+
+    EXPECT_EQ(std::nullopt, team.goalie());
+
+    team.assignGoalie(0);
+
+    EXPECT_EQ(robot_0, team.goalie());
+
+    team.clearGoalie();
+
+    EXPECT_EQ(std::nullopt, team.goalie());
+}
+
+TEST_F(TeamTest, equality_operators)
+{
+    Robot robot_0 = Robot(0, Point(0, 1), Vector(-1, -2), Angle::half(),
+                          AngularVelocity::threeQuarter(), current_time);
+
+    Robot robot_1 = Robot(1, Point(3, -1), Vector(), Angle::zero(),
+                          AngularVelocity::zero(), current_time);
+
+    Robot robot_2 = Robot(2, Point(), Vector(-0.5, 4), Angle::quarter(),
+                          AngularVelocity::half(), current_time);
+
+    Team team_0 = Team(milliseconds(1000));
+    team_0.updateRobots({robot_0, robot_2}, current_time);
+    team_0.assignGoalie(0);
+
+    Team team_1 = Team(milliseconds(1000));
+    team_1.updateRobots({robot_0, robot_2}, current_time);
+    team_1.assignGoalie(2);
+
+    Team team_2 = Team(milliseconds(1000));
+    team_2.updateRobots({robot_0, robot_1, robot_2}, current_time);
+    team_2.assignGoalie(0);
+
+    EXPECT_EQ(Team(milliseconds(1)), Team(milliseconds(1)));
+    EXPECT_EQ(team_0, team_0);
+    EXPECT_EQ(team_2, team_2);
+
+    EXPECT_NE(team_0, team_1);
+    EXPECT_NE(team_0, team_2);
+    EXPECT_NE(team_1, team_2);
+
+    // Now that team_0 and team_1 have the same goalie (they already had the same robots)
+    // they are equal. Even though team_1 was created/updated after team_0, the different
+    // timestamps are not considered for equality
+    team_0.assignGoalie(2);
+    EXPECT_EQ(team_0, team_1);
+}
```

Comment:
I think we could break up this test into several cases?

Reply:
done